### ISSUE TITLE
Validate use of $0 only if it's "simple"

### ISSUE THREAD

**Issue by chriseidhof** [realm/SwiftLint#70]
e.g. if it's the first thing that's switched on, or if the closure is really simple (not many tokens).

**Comment by jpsim** [comment]
:+1:

**Comment by segiddins** [comment]
Maybe only allow for single-line closures by default?

**Comment by chriseidhof** [comment]
Yeah... I often do this pattern:

```swift
blablaFunc {
    switch $0 {
        // multiple cases
    }
}
```

But maybe that's very specific. I guess the important thing is that the `$0` is very close to the corresponding opening brace. But having them only in single-line closures is a good first step.

**Comment by keith** [comment]
I've implemented a rule for this, but it's pretty simple. It just validates that all `$0`s have a `{` before them and a `}` after on the same line. This does mean that it doesn't catch all cases, something like:

```swift
foo.map {
     bar.something { $0 }
}
```

Wouldn't fail, but we feel like most of those cases will be caught by our line length as well. I can submit this upstream if the leniency is alright.

**Comment by jpsim** [comment]
@keith could you make a PR with your single-line rule? I can't guarantee we'd merge it right away, but it would be a good starting point, at least to stir a conversation.

**Comment by keith** [comment]
Yep, I'll do that soon.

**Comment by keith** [comment]
Submitted: https://github.com/realm/SwiftLint/pull/218
